src/Bridges/bridge_optimizer.jl

```diff
@@ -545,28 +545,67 @@ function _delete_variables_in_variables_constraints(
     end
 end
 
+function _delete_variables_in_bridged_objective(
+    b::AbstractBridgeOptimizer,
+    vis::Vector{MOI.VariableIndex},
+)
+    F = MOI.get(b, MOI.ObjectiveFunctionType())
+    _delete_variables_in_bridged_objective(b, vis, F)
+    return
+end
+
+function _delete_variables_in_bridged_objective(
+    b::AbstractBridgeOptimizer,
+    vis::Vector{MOI.VariableIndex},
+    ::Type{MOI.VectorOfVariables},
+)
+    obj_f = MOI.get(b, MOI.ObjectiveFunction{MOI.VectorOfVariables}())
+    discard = Base.Fix2(in, vis)
+    if !any(discard, obj_f.variables)
+        # There are no variables in the objective function to delete, so we
+        # don't need to do anything.
+        return
+    end
+    new_obj_f = MOI.VectorOfVariables(filter(!discard, obj_f.variables))
+    if MOI.output_dimension(new_obj_f) == 0
+        # We've deleted all variables in the objective. Zero the objective by
+        # setting FEASIBILITY_SENSE, but restore the sense afterwards.
+        sense = MOI.get(b, MOI.ObjectiveSense())
+        MOI.set(b, MOI.ObjectiveSense(), MOI.FEASIBILITY_SENSE)
+        MOI.set(b, MOI.ObjectiveSense(), sense)
+    else
+        MOI.set(b, MOI.ObjectiveFunction{MOI.VectorOfVariables}(), new_obj_f)
+    end
+    return
+end
+
+function _delete_variables_in_bridged_objective(
+    b::AbstractBridgeOptimizer,
+    vis::Vector{MOI.VariableIndex},
+    ::Type{MOI.VariableIndex},
+)
+    obj_f = MOI.get(b, MOI.ObjectiveFunction{MOI.VariableIndex}())
+    if obj_f in vis
+        sense = MOI.get(b, MOI.ObjectiveSense())
+        MOI.set(b, MOI.ObjectiveSense(), MOI.FEASIBILITY_SENSE)
+        MOI.set(b, MOI.ObjectiveSense(), sense)
+    end
+    return
+end
+
+function _delete_variables_in_bridged_objective(
+    ::AbstractBridgeOptimizer,
+    ::Vector{MOI.VariableIndex},
+    ::Type{F},
+) where {F}
+    # Nothing to do here. The variables can be deleted without changing the type
```

Comment:
I think that happens automatically. This is just a special case where deleting a variable can unset the objective, rather than just modify the existing function.

Reply:
I don't think it does. I think the variable deletion API kind of assumes that all bridges are affine so since the deletion commutes with affine operations then it's fine to skip the bridges and delete the solver model directly but there is one notable exception which is the QuadtoSOC bridge. We should maybe check but it's best we open a separate issue